pkg/controller/rds/dbcluster/setup.go

```diff
@@ -116,6 +118,29 @@ func (e *custom) postCreate(ctx context.Context, cr *svcapitypes.DBCluster, _ *s
 	}, nil
 }
 
+func isUpToDate(cr *svcapitypes.DBCluster, out *svcsdk.DescribeDBClustersOutput) (bool, error) {
+	status := aws.StringValue(out.DBClusters[0].Status)
+	if status == "modifying" || status == "upgrading" || status == "configuring-iam-database-auth" {
+		return true, nil
+	}
+
+	if aws.BoolValue(cr.Spec.ForProvider.EnableIAMDatabaseAuthentication) != aws.BoolValue(out.DBClusters[0].IAMDatabaseAuthenticationEnabled) {
+		return false, nil
+	}
+
+	return true, nil
+}
+
+func preUpdate(_ context.Context, cr *svcapitypes.DBCluster, obj *svcsdk.ModifyDBClusterInput) error {
+	obj.DBClusterIdentifier = aws.String(meta.GetExternalName(cr))
+
+	if aws.BoolValue(cr.Spec.ForProvider.ApplyImmediately) {
+		obj.ApplyImmediately = cr.Spec.ForProvider.ApplyImmediately
+	}
```

Comment:
```suggestion
		obj.ApplyImmediately = cr.Spec.ForProvider.ApplyImmediately
```
Does that make a difference?